com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs

```diff
@@ -134,103 +285,78 @@ public virtual SensorCompressionType GetCompressionType()
         }
 
         /// <summary>
-        /// Evaluates a perception vector to be used as part of an observation of an agent.
-        /// Each element in the rayAngles array determines a sublist of data to the observation.
-        /// The sublist contains the observation data for a single cast. The list is composed of the following:
-        /// 1. A one-hot encoding for detectable objects. For example, if detectableObjects.Length = n, the
-        ///    first n elements of the sublist will be a one-hot encoding of the detectableObject that was hit, or
-        ///    all zeroes otherwise.
-        /// 2. The 'length' element of the sublist will be 1 if the ray missed everything, or 0 if it hit
-        ///    something (detectable or not).
-        /// 3. The 'length+1' element of the sublist will contain the normalised distance to the object hit, or 1 if
-        ///    nothing was hit.
+        /// Evaluates the raycasts to be used as part of an observation of an agent.
+        /// </summary>
+        /// <param name="input">Input defining the rays that will be cast.</param>
+        /// <param name="output">Output class that will be written to with raycast results.</param>
+        /// <param name="debugInfo">Optional debug information output, only used by RayPerceptionSensor.</param>
         ///
+        public static void PerceiveStatic(RayPerceptionInput input, RayPerceptionOutput output)
```

Comment:
This is the public utility method for users that want to recreate the raycast logic.